includes/data-port/models/class-sensei-import-lesson-model.php

```diff
@@ -0,0 +1,496 @@
+	 */
+	private function get_quiz_args() {
+		$args = [
+			'post_type'    => 'quiz',
+			'post_parent'  => $this->get_post_id(),
+			'post_content' => '',
+		];
+
+		if ( $this->is_new() ) {
+			$current_user        = get_current_user_id();
+			$args['post_author'] = $current_user ? $current_user : $this->get_default_author();
+		} else {
+			$args['ID'] = Sensei()->lesson->lesson_quizzes( $this->get_post_id() );
+		}
+
+		$value = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_STATUS );
+		if ( null !== $value ) {
+			$args['post_status'] = $value;
+		}
+
+		$value = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_TITLE );
+		if ( null !== $value ) {
+			$args['post_title'] = $value;
+		}
+
+		$meta = $this->get_quiz_meta();
+
+		if ( is_wp_error( $meta ) ) {
+			return $meta;
+		}
+
+		$args['meta_input'] = $meta;
+
+		return $args;
+	}
+
+	/**
+	 * Helper method to get quiz meta arguments.
+	 *
+	 * @return array|WP_Error  The arguments on success, WP_Error on failure.
+	 */
+	private function get_quiz_meta() {
+		$meta = [ '_quiz_lesson' => $this->get_post_id() ];
+
+		$pass_required = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_PASS_REQUIRED );
+		$pass_mark     = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_PASSMARK );
+
+		if (
+			( empty( $pass_required ) && ! empty( $pass_mark ) ) ||
+			( ! empty( $pass_required ) && empty( $pass_mark ) ) ) {
+			return new WP_Error(
+				'sensei_data_port_passmark_incorrect',
+				__( 'Both Passmark and Pass Required should be supplied.', 'sensei-lms' )
+			);
+		}
+
+		if ( null !== $pass_required ) {
+			if ( true === $pass_required ) {
+				$meta['_pass_required'] = 'on';
+				$meta['_quiz_passmark'] = $pass_mark;
+			} else {
+				$meta['_pass_required'] = '';
+				$meta['_quiz_passmark'] = 0;
+			}
+		}
+
+		$value = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_NUM_QUESTIONS );
+		if ( null !== $value ) {
+			$meta['_show_questions'] = $value;
+		}
+
+		$value = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_RANDOMIZE );
+		if ( null !== $value ) {
+			$meta['_random_question_order'] = true === $value ? 'yes' : 'no';
+		}
+
+		$value = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_AUTO_GRADE );
+		if ( null !== $value ) {
+			$meta['_quiz_grade_type'] = true === $value ? 'auto' : 'manual';
+		}
+
+		$value = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_QUIZ_RESET );
+		if ( null !== $value ) {
+			$meta['_enable_quiz_reset'] = true === $value ? 'on' : '';
+		}
+
+		return $meta;
+	}
+
+	/**
+	 * Helper method to parse and store the quiz questions.
+	 *
+	 * @param int $quiz_id  The quiz id.
+	 *
+	 * @return bool|WP_Error  True on success, WP_Error on failure.
+	 */
+	private function set_quiz_questions( $quiz_id ) {
+		$questions = $this->get_value( Sensei_Data_Port_Lesson_Schema::COLUMN_QUESTIONS );
+		if ( null === $questions ) {
+			return true;
+		}
+
+		$this->delete_quiz_questions( $quiz_id );
```

Comment:
This feels a bit brute force. How difficult would be be to just delete the ones that aren't included? If so, we'd need to update the `_quiz_has_questions` and `_question_order` metas when no questions.

Reply:
I improved the solution by deleting all meta only when the question column is empty and by bailing early when there is no update. 39e6708ecfc7e1605f66e31b875114a854e5dc2a

There problem with this one are the question metas `_quiz_id` and `_quiz_question_order`. If we want to apply only the required modifications we need to maintain two lists (one with the questions added and one with the ones removed) and compare these with the existing meta and update accordingly. I think that this could be a bit complicated and not sure if there is much performance benefit. WDYT?

Reply:
Performance wise, not much benefit. I guess I was thinking if we re-import the same file twice, nothing should really change. How it is now, the meta IDs will continue to increment because the old ones aren't recycled. I'm perfectly fine with updating the meta when it doesn't need it, but I'd rather only delete the meta when the question is no longer included in the import. Could we grab the question IDs (from `_question_order` even) and then unset as we're looping through and then delete `_quiz_id` and `_quiz_question_order{id}` from the ones that no longer appear?

Reply:
I think it wasn't complicated after all. Implemented in 3cd7bb45c678feb0ed79674d84f77e1a220c6c79